libraries/AP_Arming/AP_Arming.cpp

```diff
@@ -493,8 +493,16 @@ bool AP_Arming::arm_checks(uint8_t method)
 {
     if ((checks_to_perform & ARMING_CHECK_ALL) ||
         (checks_to_perform & ARMING_CHECK_LOGGING)) {
-        if (!DataFlash_Class::instance()->logging_started()) {
+        // allow logging to start by indicating (falsely) that the vehicle is armed:
+        DataFlash_Class *df = DataFlash_Class::instance();
+        df->set_vehicle_armed(true);
```

Comment:
I agree that disarming after isn't a good idea, I wasn't suggesting that. I don't have a better way (yet), but I really think this isn't a great idea - your finding with Copter is what I'm afraid about, someone will break this some day. By the way, unless I'm missing something, Sub also doesn't go through here.